crates/net/discv5/src/config.rs

```diff
@@ -0,0 +1,337 @@
+        let Self {
+            discv5_config,
+            bootstrap_nodes,
+            fork,
+            tcp_port,
+            other_enr_data,
+            allow_no_tcp_discovered_nodes,
+            self_lookup_interval,
+            filter_discovered_peer,
+        } = self;
+
+        let discv5_config = discv5_config
+            .unwrap_or_else(|| discv5::ConfigBuilder::new(ListenConfig::default()).build());
+
+        let fork = fork.unwrap_or((ChainRef::ETH, MAINNET.latest_fork_id()));
+
+        let tcp_port = tcp_port.unwrap_or(DEFAULT_DISCOVERY_PORT);
+
+        let self_lookup_interval =
+            self_lookup_interval.unwrap_or(DEFAULT_SECONDS_SELF_LOOKUP_INTERVAL);
+
+        Config {
+            discv5_config,
+            bootstrap_nodes,
+            fork,
+            tcp_port,
+            other_enr_data,
+            allow_no_tcp_discovered_nodes,
+            self_lookup_interval,
+            filter_discovered_peer,
+        }
+    }
+}
+
+/// Config used to bootstrap [`discv5::Discv5`].
+#[derive(Debug)]
+pub struct Config<Filter = MustNotIncludeChains> {
+    /// Config used by [`discv5::Discv5`]. Contains the [`ListenConfig`], with the discovery listen
+    /// socket.
+    pub(super) discv5_config: discv5::Config,
+    /// Nodes to boot from.
+    pub(super) bootstrap_nodes: HashSet<BootNode>,
+    /// [`ForkId`] to set in local node record.
+    pub(super) fork: (&'static [u8], ForkId),
+    /// Mempool TCP port to advertise.
+    pub(super) tcp_port: u16,
+    /// Additional kv-pairs to include in local node record.
+    pub(super) other_enr_data: Vec<(&'static str, Bytes)>,
+    /// Allow no TCP port advertised by discovered nodes. Disallowed by default.
+    pub(super) allow_no_tcp_discovered_nodes: bool,
+    /// Interval in seconds at which to run a lookup up query with local node ID as target, to
+    /// populate kbuckets.
+    pub(super) self_lookup_interval: u64,
+    /// Optional filter rules to apply to a discovered peer in order to determine if it should be
+    /// passed up to mempool or dropped.
+    pub(super) filter_discovered_peer: Filter,
+}
+
+impl Config {
+    /// Returns a new [`ConfigBuilder`].
+    pub fn builder() -> ConfigBuilder {
+        ConfigBuilder::default()
+    }
+}
+
+impl<T> Config<T> {
+    /// Returns the discovery (UDP) socket contained in the [`discv5::Config`]. Returns the IPv6
+    /// socket, if both IPv4 and v6 are configured. This socket will be advertised to peers in the
+    /// local [`Enr`](discv5::enr::Enr).
+    pub fn discovery_socket(&self) -> SocketAddr {
+        match self.discv5_config.listen_config {
+            ListenConfig::Ipv4 { ip, port } => (ip, port).into(),
+            ListenConfig::Ipv6 { ip, port } => (ip, port).into(),
+            ListenConfig::DualStack { ipv6, ipv6_port, .. } => (ipv6, ipv6_port).into(),
+        }
+    }
+
+    /// Returns the mempool (TCP) socket contained in the [`discv5::Config`]. This socket will be
```

Comment:
noting that this is rlpx, not usually called mempool
```suggestion
    /// Returns the RLPx (TCP) socket contained in the [`discv5::Config`]. This socket will be
```